cpp/src/io/parquet/page_enc.cu

```diff
@@ -339,30 +333,27 @@ __global__ void __launch_bounds__(block_size) gpuInitPageFragments(PageFragment
         ck_row_ref = start_row + (s->dict[(hash > 0) ? s->map.u16[hash - 1] : 0] >> init_hash_bits);
         if (ck_row_ref != ck_row) {
           if (dtype == BYTE_ARRAY) {
-            const nvstrdesc_s *ck_data = static_cast<const nvstrdesc_s *>(col_data);
-            const char *str1           = ck_data[ck_row].ptr;
-            uint32_t len1              = (uint32_t)ck_data[ck_row].count;
-            const char *str2           = ck_data[ck_row_ref].ptr;
-            uint32_t len2              = (uint32_t)ck_data[ck_row_ref].count;
-            is_dupe                    = nvstr_is_equal(str1, len1, str2, len2);
-            dupe_data_size += (is_dupe) ? 4 + len1 : 0;
+            auto str1 = s->col.leaf_column->element<string_view>(ck_row);
+            auto str2 = s->col.leaf_column->element<string_view>(ck_row_ref);
+            is_dupe   = (str1 == str2);
+            dupe_data_size += (is_dupe) ? 4 + str1.length() : 0;
```

Comment:
```suggestion
            dupe_data_size += (is_dupe) ? 4 + str1.size_bytes() : 0;
```
The `string_view::length()` member is the number of characters and not the number of bytes. The number of characters in UTF-8 may not match the number of bytes. 
https://docs.rapids.ai/api/libcudf/stable/classcudf_1_1string__view.html#a7e263ef5934bdd0934bfbe5a023d88bb